nengo/cache.py

```diff
@@ -327,40 +341,42 @@ def __getitem__(self, key):
             return super(WriteableCacheIndex, self).__getitem__(key)
 
     def __setitem__(self, key, value):
+        if not isinstance(value, tuple) or len(value) != 3:
+            raise ValueError(
+                "Cache entries must include filename, start, and end.")
         self._updates[key] = value
 
     def __delitem__(self, key):
         self._deletes.add(key)
 
-    def remove_file_entry(self, filename):
-        """Remove entries mapping to ``filename``."""
-        self._removed_files.add(filename)
-
     def __enter__(self):
         with self._lock:
             try:
-                self._load_index()
+                self._load_index_file()
             except:
-                self._reinit()
+                # If we can't load the index file, the cache is corrupted,
+                # so we invalidate it (delete all files in the cache)
+                for f in os.listdir(self.cache_dir):
+                    path = os.path.join(self.cache_dir, f)
+                    if path == self._lock.filename:
+                        continue
+                    if os.path.isdir(path):
+                        shutil.rmtree(path)
+                    else:
+                        os.remove(path)
                 self._index = {}
+                self._write_index()
```

Comment:
I'm opposed to inlining this code. A function preferrably have one level of abstraction which is violated with the inlined code and makes it harder to understand. In the orginal version the code tells me immediatly: “try to load the index and if that fails reinitialize the cache" and I have only to look at the details of how the cache is reinitialized if that is actually relevant to me. With the inlined code it is not immediatly clear if all the code reinitializes the cache or whether part of it is maybe for some other error handling that is required in addition.

Especially with complex code like here, where there are a lot of things to keep in mind due to potential concurrent access, it seems helpful for me to try to keep to one level of abstraction per function. It makes it easier to reason what should happen in each function.